packages/unpublished-adapter-encapsulation-test-app/tests/integration/queries-test.js

```diff
@@ -0,0 +1,222 @@
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import EmberObject from '@ember/object';
+import Store from 'adapter-encapsulation-test-app/services/store';
+import Model, { attr } from '@ember-data/model';
+import { resolve } from 'rsvp';
+
+class MinimalSerializer extends EmberObject {
+  normalizeResponse(_, __, data) {
+    return data;
+  }
+
+  serialize(snapshot) {
+    return {
+      data: {
+        id: snapshot.id,
+        type: snapshot.modelName,
+        attributes: snapshot.attributes(),
+      },
+    };
+  }
+}
+
+class Person extends Model {
+  @attr
+  firstName;
+
+  @attr
+  lastName;
+}
+
+module('integration/queries - Queries Tests', function(hooks) {
+  setupTest(hooks);
+
+  hooks.beforeEach(function() {
+    this.owner.register('service:store', Store);
+    this.owner.register('serializer:application', MinimalSerializer);
+    this.owner.register('model:person', Person);
+  });
+
+  test('store.findRecord calls adapter.findRecord w/correct args', async function(assert) {
+    let findRecordCalled = 0;
+    let expectedResult = {
+      data: {
+        id: '12',
+        type: 'person',
+        attributes: {
+          firstName: 'Gaurav',
+          lastName: 'Munjal',
+        },
+      },
+    };
+    let { owner } = this;
+    let store = owner.lookup('service:store');
+
+    class TestFindRecordAdapter extends EmberObject {
+      findRecord(passedStore, type, id, snapshot) {
+        findRecordCalled++;
+
+        assert.equal(passedStore, store, 'instance of store is passed to findRecord');
+        assert.equal(type, Person, 'model is passed to findRecord');
+        assert.equal(id, '12', 'id is passed to findRecord');
+
+        assert.equal(snapshot.modelName, 'person', 'snapshot is passed to findRecord with correct modelName');
+        assert.equal(snapshot.id, '12', 'snapshot is passed to findRecord with correct id');
+
+        return resolve(expectedResult);
+      }
+    }
+
+    owner.register('adapter:application', TestFindRecordAdapter);
+
+    let record = await store.findRecord('person', '12');
+
+    // This code is a workaround for issue https://github.com/emberjs/data/issues/6758
+    // expectedResult is unexpectedly mutated during store.findRecord
+    // if IDENTIFIERS is turned on
+    delete expectedResult.data.lid;
+
+    assert.equal(findRecordCalled, 1, 'findRecord is called once');
+    assert.deepEqual(record.serialize(), expectedResult, 'findRecord returns expected result');
```

Comment:
If we don't re-use expectedResult we won't hit the `lid` issue.

While it's good we'e surfaced the mutation issue for discussion, you never want the `expected` to be something you've potentially operated on.

Let's refactor this to either inline the expected result or use the deep-copy util to preserve a copy when we define it. (same for the similar things below)